oxidation/libparsec/crates/types/src/shamir.rs

```diff
@@ -100,3 +108,46 @@ impl_transparent_data_format_conversion!(
     reveal_token_share,
     data_key_share,
 );
+
+impl ShamirRecoveryShareData {
+    pub fn dump(&self) -> Result<Vec<u8>, &'static str> {
+        ::rmp_serde::to_vec_named(self).map_err(|_| "Serialization failed")
```

Comment:
Use an enum instead of a raw string, it simplify testing

Reply:
a865d44fa